src/test/java/com/faforever/api/clan/ClanControllerIntegrationTest.java

```diff
@@ -81,205 +56,159 @@ public void setUp() {
         .webAppContextSetup(context)
         .addFilter(springSecurityFilterChain)
         .build();
+    database.assertEmptyDatabase();
   }
 
-  @After
-  public void tearDown() {
-    clanMembershipRepository.deleteAll();
-    clanRepository.deleteAll();
-    userRepository.deleteAll();
-    oAuthClientRepository.deleteAll();
-    assertEquals(0, clanMembershipRepository.count());
-    assertEquals(0, clanRepository.count());
-    assertEquals(0, userRepository.count());
-    assertEquals(0, oAuthClientRepository.count());
-  }
-
-  public String createUserAndGetAccessToken(String login, String password) throws Exception {
-    OAuthClient client = new OAuthClient()
-        .setId(OAUTH_CLIENT_ID)
-        .setName("test")
-        .setClientSecret(OAUTH_SECRET)
-        .setRedirectUris("test")
-        .setDefaultRedirectUri("test")
-        .setDefaultScope("test")
-        .setClientType(ClientType.PUBLIC);
-    oAuthClientRepository.save(client);
-
-    User user = (User) new User()
-        .setPassword(shaPasswordEncoder.encodePassword(password, null))
-        .setLogin(login)
-        .setEmail(login + "@faforever.com");
-    userRepository.save(user);
-    me = playerRepository.findOne(user.getId());
-
-    String authorization = "Basic "
-        + new String(Base64Utils.encode((OAUTH_CLIENT_ID + ":" + OAUTH_SECRET).getBytes()));
-    ResultActions auth = mvc
-        .perform(
-            post("/oauth/token")
-                .header("Authorization", authorization)
-                .param("username", login)
-                .param("password", password)
-                .param("grant_type", "password"));
-    auth.andExpect(status().isOk());
-    JsonNode node = objectMapper.readTree(auth.andReturn().getResponse().getContentAsString());
-    return "Bearer " + node.get("access_token").asText();
-  }
-
-  private Player createPlayer(String login) throws Exception {
-    User user = (User) new User()
-        .setPassword("foo")
-        .setLogin(login)
-        .setEmail(login + "@faforever.com");
-    userRepository.save(user);
-    return playerRepository.findOne(user.getId());
-  }
-
-
   @Test
   public void meDataWithoutClan() throws Exception {
-    String accessToken = createUserAndGetAccessToken("Dragonfire", "foo");
+    Session session = SessionFactory.createUserAndGetAccessToken(
+        database, mvc);
+
     String expected = String.format("{\"player\":{\"id\":%s,\"login\":\"%s\"},\"clan\":null}",
-        me.getId(),
-        me.getLogin());
+        session.getPlayer().getId(),
+        session.getPlayer().getLogin());
 
-    assertEquals(1, playerRepository.count());
+    assertEquals(1, database.getPlayerRepository().count());
     this.mvc.perform(get("/clans/me/")
-        .header("Authorization", accessToken))
+        .header("Authorization", session.getToken()))
         .andExpect(content().string(expected))
```

Comment:
If I touch this we should adapt it everywhere:
https://github.com/FAForever/faf-java-api/issues/69